src/lib/Stream.ts

```diff
@@ -309,6 +309,46 @@ export interface ReadableStream<T> extends Readable<T>, CommonStream<T> {
 		aborter?: (error: Error) => void
 	): ReadableStream<T>;
 
+	/**
+	 * Transform a stream into a readable stream that emits arrays of elements in
+	 * regular batches.
+	 *
+	 * The stream keeps an internal queue. When the queue reaches `maxBatchSize`, all its
+	 * contents are written to the stream.
+	 *
+	 * If `minBatchSize` is set, the queue will try to write its contents to the stream
+	 * when the queue reaches `minBatchSize`, but only if there is no previous asynchronous
+	 * write in progress.
+	 *
+	 * In other words, `minBatchSize` declares the smallest batch that successive transforms
+	 * consider "worthwhile" to handle, while `maxBatchSize` declares the largest batch that
+	 * successive transforms can comfortably handle in a single operation (think writes to a
+	 * database or calls to an external API).
+	 *
+	 * If `flushTimeout` is set, the queue will always begin writing its contents within the
+	 * specified time period (in milliseconds). In the case of sources that provide sporadic
+	 * input over a long interval, this can be used to ensure that short bursts of input
+	 * aren't held up indefinitely as they wait for enough elements to cross the `minBatchSize`
+	 * threshold.
+	 *
+	 * When the stream ends, all remaining elements in the queue are written, regardless of any
+	 * thresholds.
+	 *
+	 * @param maxBatchSize The maximum length of batch for the stream to emit.
+	 * @param options.minBatchSize? The minimum length of batch that the stream will try to emit if
+	 * no write is in progress. Defaults to `maxBatchSize`.
+	 * @param options.flushTimeout? The interval in milliseconds after which the stream will emit
+	 * all source values, even if there are fewer than `minBatchSize` of them.
+	 * @return New readable stream emitting values from its source in batches.
+	 */
+	batch(
+		maxBatchSize: number,
+		options?: {
```

Comment:
It certainly could be.
My reasoning was that the most common use case for this would be forming batches of a given size from a stream, and it's easier to call `batch(100)` than `batch({maxBatchSize: 100})`.
The other two options are significantly more specialized, I think.

I could go either way on this one.

Reply:
Ok. I saw that you took a different approach with the actual implementation function, with 'normal' arguments. I agree that `batch(100)` is the common typical case indeed.